core/trino-main/src/main/java/io/trino/sql/planner/PlanFragment.java

```diff
@@ -88,6 +90,7 @@ private PlanFragment(
         this.activeCatalogs = requireNonNull(activeCatalogs, "activeCatalogs is null");
         this.languageFunctions = requireNonNull(languageFunctions, "languageFunctions is null");
         this.jsonRepresentation = Optional.empty();
+        this.containsTableScanNode = partitionedSourceNodes.stream().anyMatch(TableScanNode.class::isInstance);
```

Comment:
Can we use List instead of Set for `partitionedSourceNodes` instead of this change to make iteration cheaper ? I don't see any utilization of set properties in the callers.

Reply:
”I don't see any utilization of set properties in the callers.“
--------
No, it does not.
But I think the Set shows a implicit constraint that the 'partitionedSourceNodes' should and would never contains duplicate Node, so it is better for use Set instead of List.
And, iteration there would called only one times for each fragment, so the performance of iterate here does not very matter anymore.